lib/classes/ConfigSchemaHandler/index.js

```diff
@@ -290,6 +309,21 @@ class ConfigSchemaHandler {
 }
 
 function addPropertiesToSchema(subSchema, extension = { properties: {}, required: [] }) {
+  let collidingExtensionPropertyKey;
+  const existingSubSchemaPropertiesKeys = Object.keys(subSchema.properties);
+  Object.keys(extension.properties).some(extensionPropertiesKey => {
```

Comment:
At first I was thinking why not to throw an error inside the callback, but I've noticed you used `some` to stop iteration on first colliding property, nice :+1: